utility/import_exporter.cpp

```diff
@@ -0,0 +1,843 @@
+		if (path_found && lossy_texture){
+			String basedir = i_info->import_path.get_base_dir();
+			Vector<String> split = i_info->import_path.get_file().split(".");
+			if (split.size() == 4) {
+				// for example, "res://.import/Texture.png-cefbe538e1226e204b4081ac39cf177b.s3tc.stex"
+				// will become "res://.import/Texture.png-cefbe538e1226e204b4081ac39cf177b.stex"
+				String new_path = basedir.plus_file(split[0] + "." + split[1] + "." + split[3]);
+				// if we have the ghost .stex, set it to be the import path
+				if (GDRESettings::get_singleton()->has_res_path(new_path)) {
+					i_info->import_path = new_path;
+				}
+			}
+		}
+	}
+
+	ERR_FAIL_COND_V_MSG(!path_found || i_info->type == String(), ERR_FILE_CORRUPT, p_path + ": file is corrupt");
+	if (cf->has_section("params")){
+		List<String> param_keys; 
+		cf->get_section_keys("params", &param_keys);
+		i_info->params = Dictionary();
+		for (auto E = param_keys.front(); E; E = E->next()){
+			i_info->params[E->get()] = cf->get_value("params", E->get(), "");
+		}
+	}
+	i_info->cf = cf;
+	
+	files.push_back(i_info);
+	return OK;
+}
+
+// export all the imported resources
+Error ImportExporter::export_imports(const String &output_dir){
+	String out_dir = output_dir == "" ? output_dir : GDRESettings::get_singleton()->get_project_path();
+	Error err = OK;
+	if (opt_lossy) {
+		WARN_PRINT_ONCE("Converting lossy imports, you may lose fidelity for indicated assets when re-importing upon loading the project");
+	}
+
+	for (int i = 0; i < files.size(); i++){
+		Ref<ImportInfo> iinfo = files[i];
+		String path = iinfo->get_path();
+		String source = iinfo->get_source_file();
+		String type = iinfo->get_type();
+		String importer = iinfo->importer;
+		auto loss_type = iinfo->get_import_loss_type();
+		if (loss_type != ImportInfo::LOSSLESS){
+			if (!opt_lossy){
+				lossy_imports.push_back(iinfo);
+				print_line("Not converting lossy import " + path);
+				continue;
+			}
+		}
+		if (opt_export_textures && importer == "texture"){
+			// Right now we only convert 2d image textures
+			auto tex_type = TextureLoaderCompat::recognize(path, &err);
+			switch (tex_type){
+				case TextureLoaderCompat::FORMAT_V2_IMAGE_TEXTURE:
+				case TextureLoaderCompat::FORMAT_V3_STREAM_TEXTURE2D:
+				case TextureLoaderCompat::FORMAT_V4_STREAM_TEXTURE2D: {
+					// Export texture
+					err = export_texture(out_dir, iinfo);
+				} break;
+				case TextureLoaderCompat::FORMAT_NOT_TEXTURE:
+					if (err == ERR_FILE_UNRECOGNIZED) {
+						WARN_PRINT("Import of type " + type + " is not a texture(?): " + path);
+					} else {
+						WARN_PRINT("Failed to load texture " + type + " " + path);
+					}
+					break;
+				default:
+					WARN_PRINT_ONCE("Conversion for " + type + " not yet implemented");
+					print_line("Did not convert " + type + " resource " + path);
+					not_converted.push_back(iinfo);
+					continue;
+			}
+		} else if (opt_export_samples && importer == "sample" || importer == "wav") {
```

Comment:
```suggestion
		} else if (opt_export_samples && (importer == "sample" || importer == "wav")) {
```
Assuming, `opt_export_samples` should be set for both `wav` and `sample`.